docs/custom-rules.md

```diff
@@ -164,6 +164,83 @@ Starting from top to bottom, these are the components comprising our custom rule
    will later be included in the final report provided by Regal.
 1. The `result.location` helps extract the location from the element failing the test. Make sure to use it!
 
+## Aggregate Rules
+
+Aggregate rules are a special type of rule that allows you to collect data from multiple files before making a decision.
+This is needed in some cases where a single policy won't be enough for a linter rule to make a decision. For example,
+you may want to enforce that in a policy repository, there must be at least one package annotated with an `authors`
+attribute. This requires first collecting annotation data from all the provided policies, and then have the rule use
+this data to make a decision. The structure of an aggregate rule is similar to a regular rule, but with a few notable
+differences. The most significant one is that linting happens in two separate phases — one that aggregates data from
+files, and one that actually lints and reports violations using that data.
```

Comment:
Take or leave this one, I thought it made sense to strictly tie the explanation to the two rules in the code below.

```suggestion
files (`aggregate`), and one that actually lints and reports violations using that data (`aggregate_report`).
```